### PR TITLE
Swift Basic/Driver recognizes OpenBSD.

### PR DESCRIPTION
**Pull request by 3405691582**
Add the platform conditional and set up other basics for the toolchain.
    
The ConditionalCompilation tests are updated to match, since otherwise
they seem to trip when building on non-OpenBSD platforms.

This is orthogonal to pr #30082, which needs to wait for pending build system changes. This can proceed independently, but can wait if necessary.

### CODE REVIEW

File: lib/Driver/ToolChains.h [hunk 1]
```diff
@@ -157,6 +157,17 @@ class LLVM_LIBRARY_VISIBILITY Cygwin : public GenericUnix {
   ~Cygwin() = default;
 };
 
+class LLVM_LIBRARY_VISIBILITY OpenBSD : public GenericUnix {
+protected:
+  std::string getDefaultLinker() const override;
+
```
Comment (compnerd) [inline lib/Driver/ToolChains.h]: Unnecessary whitespace

**Review by compnerd — APPROVED** [review]
I think it would be nice to have a test to ensure that `lld` is invoked by default for OpenBSD.

File: lib/Driver/ToolChains.h (hunk 1 above)
Comment (3405691582) [reply lib/Driver/ToolChains.h]: Removed.

**Comment by 3405691582** [comment]
> I think it would be nice to have a test to ensure that `lld` is invoked by default for OpenBSD.

Added to Driver/linker.swift, cribbing off existing stanzas, since it was simple enough.

**Review by finagolfin — APPROVED** [review]

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by CodaFi** [comment]
Cool!

⛵️
